Automattic/wp-calypso
# Adds tracks event for action that was removed

In https://github.com/Automattic/wp-calypso/pull/29458 we removed the need for the add email button because we show the email fields automatically now.  We removed the step the tracks event was tracking. Since you cannot take that action we are firing the event on mount. This also removed the need for the tracks event.  In order for funnels to match up we need that event. 

#### Changes proposed in this Pull Request

* Adds back the `AddEmailButtonClick` call so that the tracks event is recorded
* This is added to `handleFormSubmit` function so that it is sent when the form is submitted

#### Testing instructions

* Load branch
* Goto a site
* Goto Domains
* Click "Add a domain"
* Choose any domain
* Observe that you can see the G Suite user fields
* Observe that event fires (`calypso:analytics:tracks Record event "calypso_google_apps_add_email_button_click" called with props` )

![screen shot 2019-01-03 at 11 37 02 am](https://user-images.githubusercontent.com/6817400/50649500-0d810080-0f4c-11e9-8dfb-9d53cad42464.png)

You can turn on analytics logging by running this in the console: localStorage.setItem( 'debug', 'calypso:analytics:*' )
